src/python/gennylib/genny_runner.py

```diff
@@ -0,0 +1,68 @@
+import sys
+import os
+import subprocess
+from contextlib import contextmanager
+
+def get_genny_args():
+    """
+    Returns the argument list used to create the Genny process.
+
+    If we are in the root of the genny repo, use the local executable. 
+    Otherwise we search the PATH.
+    """
+    args = sys.argv
+    genny_core = 'genny_core'
+
+    local_core = './dist/bin/genny_core'
+    if os.path.exists(local_core):
+        genny_core = local_core
+    args[0] = genny_core
+    return args
+
+def get_poplar_args():
+    """
+    Returns the argument list used to create the Poplar gRPC process.
+
+    If we are in the root of the genny repo, use the local executable. 
+    Otherwise we search the PATH.
+    """
+    curator = 'curator'
+
+    local_curator = './curator/curator'
+    if os.path.exists(local_curator):
+        curator = local_curator
+    return [curator, 'poplar', 'grpc']
+
+@contextmanager
+def poplar_grpc():
+    poplar = subprocess.Popen(get_poplar_args())
+    if poplar.poll() is not None:
+        raise OSError("Failed to start Poplar.")
+
+    try:
+        yield poplar
+    finally:
+        try:
+            poplar.terminate()
+            exit_code = poplar.wait(timeout=10)
+            if exit_code not in (0, -15): # Termination or exit.
+                raise OSError("Poplar exited with code: {code}.".format(code=(exit_code)))
+
+        except:
+            # If Poplar doesn't die then future runs can be broken.
```

Comment:
This is... alarming. What error is poplar throwing? We should file a MAKE ticket if we can't get poplar to shutdown.

Reply:
Per offline discussion, we are just cleaning up resources here. We expect terminate() to work in the happy case.